### PR TITLE
Update development guide

### PR DESCRIPTION
**Pull request by cdrage**
Updates the development guide and adds more detailed instructions on
what to look out for with reviews (submitting and reviewing).

Closes: https://github.com/redhat-developer/odo/issues/609

### CODE REVIEW

**Comment by cdrage** [comment]
Feel free to suggest more things to add @kadel / @ashetty1 !

File: docs/development.md [hunk 1]
```diff
@@ -203,8 +239,9 @@ When new git tag is created, Travis-ci deploy job automatically builds binaries
 4. Verify that packages have been uploaded to rpm and deb repositories.
 5. Confirm the binaries are available in GitHub release page and update the file `build/VERSION` with latest version number.
 
-## odo-bot
-This is GitHub user that does all the automation.
+## Odo Bot
+
+[odo-bot](https://github.com/odo-bot) is the GitHub user that provides automation for certain tasks of Odo.
```
Comment (anmolbabu) [inline docs/development.md]: Can we detail out what tasks?
At least a few examples?

Comment (kadel) [reply docs/development.md]: It is already there. Check next section "Scripts using odo-bot"

Comment (kadel) [reply docs/development.md]: @anmolbabu do you think it is enough? Or there should be more?
Feel free to merge this is you are OK with it.

**Pull request merged**